project/SparkBuild.scala

```diff
@@ -443,6 +443,7 @@ object Hive {
         val relGFilePath = (gFilePath relativeTo sourceDir).get.getPath
         log.info("ANTLR: Grammar file '%s' detected.".format(relGFilePath))
         antlr.addGrammarFile(relGFilePath)
+        antlr.setLibDirectory(gFilePath.getParent)
```

Comment:
Yes, I noticed that. You are correct.

I would prefer the current one even it sets the library path multiple times and assumes there is only one library directory.

If you and @rxin both think it is better, I can change it.